src/core/operations/ECDSASignatureConversion.mjs

```diff
@@ -0,0 +1,106 @@
+/**
+ * @author cplussharp
+ * @copyright Crown Copyright 2021
+ * @license Apache-2.0
+ */
+
+import Operation from "../Operation.mjs";
+import r from "jsrsasign";
+
+/**
+ * ECDSA Sign operation
+ */
+class ECDSASignatureConversion extends Operation {
+
+    /**
+     * ECDSASignatureConversion constructor
+     */
+    constructor() {
+        super();
+
+        this.name = "ECDSA Signature Conversion";
+        this.module = "Ciphers";
+        this.description = "Convert an ECDSA signature between hex, asn1 and json.";
+        this.infoURL = "https://wikipedia.org/wiki/Elliptic_Curve_Digital_Signature_Algorithm";
+        this.inputType = "string";
+        this.outputType = "string";
+        this.args = [
+            {
+                name: "Input Format",
+                type: "option",
+                value: [
+                    "Auto",
+                    "ASN.1 HEX",
+                    "Concat HEX",
+                    "JSON"
+                ]
+            },
+            {
+                name: "Output Format",
+                type: "option",
+                value: [
+                    "ASN.1 HEX",
+                    "Concat HEX",
+                    "JSON"
+                ]
+            }
+        ];
+    }
+
+    /**
+     * @param {string} input
+     * @param {Object[]} args
+     * @returns {string}
+     */
+    run(input, args) {
+        let inputFormat = args[0];
+        const outputFormat = args[1];
+
+        // detect input format
+        if (inputFormat === "Auto") {
+            if (input.substr(0, 2) === "30" && r.ASN1HEX.isASN1HEX(input)) {
+                inputFormat = "ASN.1 HEX";
+            } else if (input.indexOf("{") !== -1) {
+                inputFormat = "JSON";
+            } else {
+                inputFormat = "Concat HEX";
+            }
+        }
+
+        // convert input to ASN.1 hex
+        let signatureASN1Hex;
+        switch (inputFormat) {
+            case "ASN.1 HEX":
+                signatureASN1Hex = input;
+                break;
+            case "Concat HEX":
```

Comment:
This seems to be overly permissive, and could lead to misleading answers.  E.g. this is an invalid entry, but gives me a sane-looking key back [example reproduction](http://localhost:8080/#recipe=ECDSA_Signature_Conversion('Concat%20HEX','ASN.1%20HEX')&input=LS0tLS1CRUdJTiBQVUJMSUMgS0VZLS0tLS0NCk1JR2JNQkFHQnlxR1NNNDlBZ0VHQlN1QkJBQWpBNEdHQUFRQUhoamNZRkNOMVJ2bFBTQUVCdldkWENEM3M4UWENCkVjeDM3dE9rVGEvUDJvaXJURngyUmlZdWhPeWhFMCtNUC9WS2JrUWNvWHVHK0lvRVZ1RDZUNmoxTHNRQTBYQTYNClMrRzNKLytFTmZHVUFVMm42R2R0dkREMWdXaWNEVlR3VmI1ajB6N2xOODd5VTgyREtCOXl5TDllS005dklWTFoNClNrRndvMEVuanJwNXI3Ujloa3M9DQotLS0tLUVORCBQVUJMSUMgS0VZLS0tLS0&ieol=CRLF&oeol=CRLF)

Reply:
I will add a check if the input is really a hex value